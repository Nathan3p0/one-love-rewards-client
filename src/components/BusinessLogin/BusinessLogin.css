.main__content-business_login {
    display: flex;
    justify-content: center;
    align-items: center;
}

.main__content-business_login h2 {
    font-size: 2.5rem;
    margin-bottom: 2rem;
    align-self: center;
}

.form__input-container {
    width: 75%;
}

.businessLoginForm {
    border-radius: 1rem;
    width: 40vw;
    height: 25vh;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

input[type="text"],
input[type="password"],
input[type="email"],
input[type="tel"]{
    width: 100%;
    border: .2rem solid rgba(0, 0, 0, 0.3);
    border-radius: .3rem;
    font-size: 1.5em;
    margin-right: .2rem;
    margin-top: 1rem;
    margin-bottom: 1rem;
    padding: .8rem 0;
}

input[type="text"]:hover,
input[type="password"]:hover {
    cursor: pointer;
    border: .2rem solid rgba(0, 162, 255, 0.849);
}

label {
    font-size: 1.5rem;
    font-weight: 500;
    margin: 2rem 0 2rem 0;
}

button.biz-btn {
    width: 15rem;
    margin-top: 2rem;
    transition: all 0.5s cubic-bezier(0.65, -0.25, 0.25, 1.95);
    font-weight: 900;
    color: #202121;
    padding: 1.25rem 2rem;
    background: #FFE74C;
    text-transform: uppercase;
}

button.biz-btn:hover,
button.biz-btn:focus,
button.biz-btn:active {
    letter-spacing: 0.125rem;
}

@media screen and (max-width: 800px) {
    .main__content-business_login {
        margin-top: 5rem;
    }

    .businessLoginForm {
        padding: 0;
        border-radius: 1rem;
        width: 100%;
        height: 100%;
        display: flex;
        justify-content: center;
        align-items: center;
        flex-direction: column;
    }
}