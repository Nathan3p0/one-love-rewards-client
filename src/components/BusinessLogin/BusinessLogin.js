import React, { Component } from 'react'
import './BusinessLogin.css'
import AuthApiService from '../../services/auth-api-service'
import TokenService from '../../services/token-service'

class BusinessLogin extends Component {
    state = {
        error: null
    }

    handleSubmitJwtAuth = e => {
        e.preventDefault()
        this.setState({ error: null })
        const { username, password } = e.target

        AuthApiService.postLogin({
            username: username.value,
            password: password.value
        })
            .then(res => {
                username.value = ''
                password.value = ''
                TokenService.saveAuthToken(res.authToken)
                this.props.onLoginSuccess()
            })
            .catch(
                res => {
                    this.setState({ error: res.error })
                }
            )
    }

    render() {
        const { error } = this.state

        return (
            <section className="main__content-business_login">
                <div>
                    <form onSubmit={this.handleSubmitJwtAuth} className="businessLoginForm">

                        <h2>Business Owner Login</h2>
                        <div role='alert'>
                            {error && <p className='red'>{error}</p>}
                        </div>
                        <div>
                            <label htmlFor="username">Username:</label>
                            <br />
                            <input type='text' id="username" name="username" placeholder="username" required />
                        </div>
                        <div>
                            <label htmlFor="password">Password:</label>
                            <br />
                            <input type="password" id="password" name="password" placeholder="password" required />
                        </div>
                        <button type="submit" className="biz-btn">Submit</button>
                    </form>
                </div>
            </section>
        )
    }
}

export default BusinessLogin