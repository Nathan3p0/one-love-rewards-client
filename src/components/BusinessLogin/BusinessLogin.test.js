import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import BusinessLogin from './BusinessLogin'

describe('Business Login Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BusinessLogin />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<BusinessLogin />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<BusinessLogin
            onLoginSuccess={() => { }}
        />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})