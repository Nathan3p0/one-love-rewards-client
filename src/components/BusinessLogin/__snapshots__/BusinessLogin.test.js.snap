// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Business Login Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="main__content-business_login"
>
  <div>
    <form
      className="businessLoginForm"
      onSubmit={[Function]}
    >
      <h2>
        Business Owner Login
      </h2>
      <div
        role="alert"
      />
      <div
        className="form__input-container"
      >
        <label
          htmlFor="username"
        >
          Username:
        </label>
        <br />
        <input
          id="username"
          name="username"
          placeholder="username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form__input-container"
      >
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <br />
        <input
          id="password"
          name="password"
          placeholder="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="biz-btn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</section>
`;

exports[`Business Login Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<section
  className="main__content-business_login"
>
  <div>
    <form
      className="businessLoginForm"
      onSubmit={[Function]}
    >
      <h2>
        Business Owner Login
      </h2>
      <div
        role="alert"
      />
      <div
        className="form__input-container"
      >
        <label
          htmlFor="username"
        >
          Username:
        </label>
        <br />
        <input
          id="username"
          name="username"
          placeholder="username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form__input-container"
      >
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <br />
        <input
          id="password"
          name="password"
          placeholder="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="biz-btn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</section>
`;
